users/common/utils.py

```diff
@@ -18,23 +21,28 @@ def generate_random_phone_number(length):
 
 
 def raise_serializer_exception(error_message):
+    logger.warning(f"Serializer exception with message: {error_message}")
     raise serializers.ValidationError(error_message)
 
 
 def raise_custom_validation_exception(error_message, error_code):
+    logger.warning(f"Custom valid exception with message: {error_message} and error code: {error_code}")
     raise CustomValidationError(error_message, error_code)
 
 
 def prepare_proper_exception(call_instance, error_message, error_code):
     if "serializers" in str(type(call_instance)):
+        logger.debug(f"Preparing exception caught in serializer instance: {call_instance}")
         raise_serializer_exception(error_message)
     else:
         raise_custom_validation_exception(error_message, error_code)
 
 
 def custom_validate_email_function(call_instance, email):
+    logger.info(f"Custom validate email function with email: {email} and call instance: {call_instance}")
```

Comment:
I don't really get what means `call_instance`. It should have more meanful name.

Reply:
In logs it return class which call this method with all fields and their details.
For example:
```
CustomRegisterSerializer(data=<QueryDict: {'csrfmiddlewaretoken': ['wU2Yw4LUrH86pnEvxvbLjfjGZVeSBviNTjzlIQIGIgGkmOTKSSUiKHcAnIS3DrGm'], 'email': ['dsa@wp.pl'], 'first_name': ['fsafas'], 'last_name': ['fasfasf'], 'password': ['123456'], 'password_confirmation': ['123456']}>):
    email = EmailField(required=True)
    first_name = CharField(allow_blank=True, required=False, write_only=True)
    last_name = CharField(allow_blank=True, required=False, write_only=True)
    password = CharField(label='Password', required=True, style={'input_type': 'password'}, write_only=True)
    password_confirmation = CharField(label='Password confirmation', required=True, style={'input_type': 'password'}, write_only=True)
```

Reply:
That is too much details, we just need class name in log.